torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -167,6 +184,13 @@ class ProcessGroupGloo : public ProcessGroup {
     // is probably cheaper (this is highly speculative).
     std::unique_ptr<::gloo::Exception> ex_;
 
+    // List of devices and events so that we can synchronize the
+    // streams of the caller with the kernels that were launched
+    // asynchronously to finish this operation.
+    bool cuda_;
+    std::vector<int> devices_;
+    std::vector<CUDAEvent> events_;
```

Comment:
Similar question like above. Would be good to have a reference to the note at the top.

Reply:
Added comment clarifying the use of local events here.